hphp/runtime/vm/jit/tc-relocate.cpp

```diff
@@ -518,18 +531,30 @@ bool relocateNewTranslation(TransLoc& loc, CodeCache::View cache,
   }
 
   if (asm_count) {
-    x64::adjustForRelocation(rel);
-    x64::adjustMetaDataForRelocation(rel, nullptr, fixups);
-    x64::adjustCodeForRelocation(rel, fixups);
+    adjustForRelocation(rel);
+    adjustMetaDataForRelocation(rel, nullptr, fixups);
+    adjustCodeForRelocation(rel, fixups);
   }
 
   if (debug) {
     auto clearRange = [](TCA start, TCA end) {
       CodeBlock cb;
       cb.init(start, end - start, "Dead code");
-      X64Assembler a {cb};
-      while (cb.available() >= 2) a.ud2();
-      if (cb.available() > 0) a.int3();
+      switch (arch()) {
+        case Arch::PPC64: {
+          ppc64_asm::Assembler a{cb};
+          a.emitTrap();
+          break;
+        }
+        case Arch::X64: {
+          X64Assembler a{cb};
+          a.emitTrap();
+          break;
+        }
+        case Arch::ARM:
+          not_implemented();
+          break;
+      }
```

Comment:
well basically we need to emit `ud2{}` vasm, but on the other hand, the designed idea is to emit traps until the end of the CodeBlock. I don't think it's possible to do it with a Vout (at least judging from all other `code-gen-helpers.h` functions, which use Vout)